primitives/runtime/src/traits.rs

```diff
@@ -540,6 +540,9 @@ morph_types! {
 	/// Morpher to disregard the source value and replace with another.
 	pub type Replace<V: TypedGet> = |_| -> V::Type { V::get() };
 
+	/// Morpher to disregard the source value and replace with default for T.
+	pub type ReplaceWithDefaultFor<T: Default> = |_| -> T { T::default() };
```

Comment:
```suggestion
	/// Morpher to disregard the source value.
	pub type Ignore = |_| -> () { () };
```
? Otherwise the usage of `MapSuccess<Self::PromoteOrigin, ReplaceWithDefaultFor<()>>` is difficult to understand IMHO.